src/mappings/fixedRateExchange.ts

```diff
@@ -176,6 +176,12 @@ export function handleSwap(event: Swapped): void {
   swap.exchangeId = fixedRateId
   swap.by = getUser(event.params.by.toHex()).id
 
+  // Track OPC swap fee
+  swap.oceanFeeAmount = weiToDecimal(
+    event.params.oceanFeeAmount.toBigDecimal(),
+    BigInt.fromI32(18).toI32()
```

Comment:
OPF Fee is always taken in baseToken, so it's not always using 18 decimals.
You should use something like:
```
swap.oceanFeeAmount= weiToDecimal(
    event.params.oceanFeeAmount.toBigDecimal(),
    BigInt.fromI32(baseToken.decimals).toI32()
  )
  ```
  and move this code after we have baseToken  (like before swap.save() )